apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -57,12 +59,18 @@ def _guess_location(item: pystac.Item) -> Tuple[str, bool]:
         # Override self with canonical
         if link.rel == "canonical":
             self_link = link.target
+            break
 
-    for name, asset in item.assets.items():
+    for _, asset in item.assets.items():
         if "geotiff" in asset.media_type:
             asset_link = os.path.dirname(asset.href)
             break
 
+    if rewrite is not None:
+        for _, asset in item.assets.items():
+            if "geotiff" in asset.media_type:
```

Comment:
I know this is used through out the code, but I think this is wrong way to go, why are we assuming that only geotiff image type will be used with STAC. I think we should look for `data` role rather than filtering on media type.
And in this specific case why not rewrite metadata urls as well as imagery urls?

Reply:
We can't just stick to `data` role, as other assets have roles like `metadata` for SCL.

In fact, the Landsat STAC isn't specifying `data` for the data bands! https://planetarycomputer.microsoft.com/api/stac/v1/collections/landsat-8-c2-l2/items

Reply:
For metadata URLs, some are direct links to .json files on a http/s URI, and may be useful there still.

Maybe it's not going to be possible to handle generically like this.